Source/JavaScriptCore/bytecode/CallLinkInfo.cpp

```diff
@@ -397,49 +398,71 @@ void CallLinkInfo::setVirtualCall(VM& vm, JSCell* owner)
 #endif
     if (isDataIC()) {
         m_calleeOrCodeBlock.clear();
-        *bitwise_cast<uintptr_t*>(m_calleeOrCodeBlock.slot()) = (bitwise_cast<uintptr_t>(owner) | polymorphicCalleeMask);
+        *bitwise_cast<uintptr_t*>(m_calleeOrCodeBlock.slot()) = polymorphicCalleeMask;
         u.dataIC.m_codeBlock = nullptr; // PolymorphicCallStubRoutine will set CodeBlock inside it.
         u.dataIC.m_monomorphicCallDestination = vm.getCTIVirtualCall(callMode()).code().template retagged<JSEntryPtrTag>();
     }
     setClearedByVirtual();
     m_mode = static_cast<unsigned>(Mode::Virtual);
 }
 
+JSGlobalObject* CallLinkInfo::globalObjectForSlowPath(JSCell* owner)
+{
+    auto [codeBlock, bytecodeIndex] = retrieveCaller(owner);
```

Comment:
Why use `retrieveCaller` when you don't need the `bytecodeIndex` result here?  Why not just do a `jsDynamicCast<CodeBlock*>(owner)`?

Reply:
`jsDynamicCast<CodeBlock*>(owner)` does not work. since DFG can have inlined CodeBlocks.

Reply:
I saw the early falsification in `retrieveCaller` but missed that later it doesn't just return the owner.